### PR TITLE
Prepare Release v2.5.0-dev.1

### LINKED ISSUE #536: Publish release with the "next" dist-tag
Let's wait until #535 is resolved to do this.

This will resolve #474.

### PR DESCRIPTION
**Pull request by adidahiya**
- Fixes #536
- Add docs about "next" distribution to README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,14 @@ TSLint
...
 A linter for the TypeScript language.
 
+"next" distribution
+-------------------
+
+The `next` [branch of the TSLint repo](https://github.com/palantir/tslint/tree/next) tracks the latest TypeScript
+compiler and allows you to lint TS code that uses the latest features of the language. Releases from this branch
+are published to npm with the `next` dist-tag, so you can get the latest dev version of TSLint via
+`npm install tslint@next`.
+
```
Comment (adidahiya) [inline README.md]: this part should probably also be added to the master branch, but I'll do that in a separate PR

**Comment by ashwinr** [comment]
how will tslint handle 2.4.0 releases?

**Comment by ashwinr** [comment]
2.4.x releases, i mean

**Comment by gscshoyru** [comment]
We still release 2.4.x release, just make sure they're tagged with the "latest" tag on npm. All of our releases for the future will be with the "next" tag, and not the latest tag. So when we release this, the "latest" on npm won't update -- same as what typescript is doing.

**Comment by ashwinr** [comment]
cool, makes sense. :+1:

**Pull request merged**